torch/csrc/autograd/engine.cpp

```diff
@@ -880,6 +880,8 @@ auto Engine::execute(const edge_list& roots,
                      bool create_graph,
                      bool accumulate_grad,
                      const edge_list& outputs) -> variable_list {
+  const auto guard = c10::impl::VirtualGuardImpl{c10::DeviceType::CUDA};
```

Comment:
What is this guard doing? Just to capture the current stream at the beginning of the call?
Why not capture that at the end where we do the rest of the stream logic?

Ho you actually already do that there. So this is dead code?

Reply:
It was just there to serve the debugging print below :P